src/dsync/dsync.c

```diff
@@ -3311,7 +3312,21 @@ int main(int argc, char **argv)
     if (rank == 0) {
         MFU_LOG(MFU_LOG_INFO, "Walking source path");
     }
-    mfu_flist_walk_param_paths(1, srcpath, walk_opts, flist_tmp_src, mfu_src_file);
+    walk_rc = mfu_flist_walk_param_paths(1, srcpath, walk_opts, flist_tmp_src, mfu_src_file);
+
+    /* If we encountered an error during the srcpath walk, the src flist is likely incomplete,
+     * and a delete might delete files already on the destination.  Disable the delete and
+     * notify the user. rsync takes the same approach. */
+    int all_rc;
+    MPI_Allreduce(&walk_rc, &all_rc, 1, MPI_INT, MPI_MAX, MPI_COMM_WORLD);
+    if (all_rc > 0) {
```

Comment:
FYI there is a helper function for this
```suggestion
    if (!mfu_alltrue(walk_rc == 0, MPI_COMM_WORLD) {
```